### DISCUSSION TITLE
Optimizing material parameters using postprocessor values

### DISCUSSION THREAD

**Discussion by andrisfreimanis** [idaholab/moose#27087] [Q&A Modules: General]
Dear All,
I am looking for a way to fit a simulated stress-strain curve to an experimental stress-strain curve by optimizing material parameters. I saw that the Optimization module has examples on material inversion, however, they optimize point-based values and I didn't see anything that would work with postprocessor (global) values.

My use-case would be a crystal plasticity simulation. For example, CrystalPlasticityKalidindiUpdate material model. I know all parameters, except exponent n and preexponent gammaDot0. At each timestep two postprocessors calculate global average stress and strain. I want to compare these calculated stress-strain values to experimental stress-strain values. Then, change the exponent and preexponent material parameters to minimize the error between simulated-experimental stress-strain values. Is something like this possible using the Optimization module? 

Best regards
Andris

**Comment by maxnezdyur** [comment]
I would look at this example: https://mooseframework.inl.gov/modules/optimization/examples/shapeOpt_Annulus.html
Ignore the parts about the constraints, but given a postprocessor that is your objective value, you can use the derivative free nelder-mead, or the finite-differencing gradient option to optimize the parameters to get you the lowest postprocessor value.
